core/src/context.rs

```diff
@@ -246,4 +250,59 @@ impl Context {
 
         Ok(())
     }
+
+    pub fn spawn<F>(&self, target: F) where F: 'static + Send + Fn(&mut ContextProxy) {
+        let mut cxp = ContextProxy {
+            current: self.current,
+            event_proxy: self.event_proxy.as_ref().map(|p| p.make_clone())
+        };
+
+        std::thread::spawn(move || target(&mut cxp));
+    }
+}
+
+/// A bundle of data representing a snapshot of the context when a thread was spawned. It supports
+/// a small subset of context operations. You will get one of these passed to you when you create a
+/// new thread with `cx.spawn()`.
+pub struct ContextProxy {
+    pub current: Entity,
+    pub event_proxy: Option<Box<dyn EventProxy>>,
+}
+
+pub enum ProxyEmitError {
+    Unsupported,
+    EventLoopClosed,
+}
+
+impl ContextProxy {
+    pub fn emit<M: Message>(&mut self, message: M) -> Result<(), ProxyEmitError> {
+        if let Some(proxy) = &self.event_proxy {
+            let event = Event::new(message)
+                .target(self.current)
+                .origin(self.current)
```

Comment:
Since the `ContextProxy` passed to the thread is mutable the user could for whatever reason change the `current` entity before sending an event, which might cause some unexpected bugs. I don't know though if this is something that should be prevented or if that would just be considered 'user error'?

Reply:
The same thing could be said about the normal Context. Should we treat this one differently?

Reply:
Fair point. Let's leave it for now then.